### DISCUSSION TITLE
Force theme variables generation

### DISCUSSION THREAD

**Discussion by mildlyfunctional** [RyanClementsHax/tailwindcss-themer#37] [Q&A]
Say I have theme my-yellow-theme. Is there a way to force the generation of the themed custom properties? Even when the .my-yellow-theme class is not in the document?

**Comment by RyanClementsHax** [comment]
Hiya! 👋

I believe that is possible but not through this package. If I am not mistaken, the css that tailwind generates is pruned using purgecss. Generating all of tailwinds classes would bloat bundle size so purgecss is used to remove all unused css. If you need to have some unused classes survive, I'd look at the tailwind docs regarding purgecss. I hope that helps.

Btw if you don’t mind me asking, what is your use case?

**Reply by mildlyfunctional** [reply]
Hi Ryan,

You just gave me the idea of trying the safelist option. I just need to keep the custom properties. Actually what I'm trying to do is to create separate files. The use case are web components. So within shadow dom the default theme is used. But the web component consumer can use a specific theme.

And thanks for this great plugin and the quick response.

**Reply by mildlyfunctional** [reply]
For your information: I was unable to use the safelist option, but I think I can use the themeUtils.resolveThemeExtensionAsCustomProps to get the custom properties.

**Reply by RyanClementsHax** [reply]
While technically you could do that, those are internals to the plugin and could change even between bug fix versions. The stable parts of the plugin are what are documented.

**Reply by mildlyfunctional** [reply]
Yes I am aware. For the moment it is just a proof of concept.
